include/seqan3/utility/container/small_string.hpp

```diff
@@ -363,6 +363,26 @@ class small_string : public small_vector<char, capacity_ + 1>
     {
         return str();
     }
+
+    /*!\brief Implicit conversion to std::string_view.
+     *
+     * It is the programmer's responsibility to ensure that the resulting string view does not outlive the string.
+     *
+     * ### Exceptions
+     *
+     * Strong exception guarantee. No data is modified.
+     *
+     * ### Complexity
+     *
+     * Constant.
+     *
+     * \experimentalapi{Experimental since version 3.1.}
+     * \sa https://en.cppreference.com/w/cpp/string/basic_string/operator_basic_string_view
+     */
+    constexpr operator std::string_view() const noexcept
```

Comment:
This will be ambiguous in C++23, because in C++23 `std::string_view` has a constructor for all contiguous char ranges (including `seqan3::small_string`).

Reply:
When I run gcc12 in C++23 mode, which has the range ctor for `string_view`, I don't get ambiguity.

```cpp
constexpr seqan3::small_string cfoo{"hello"};
consteval int constexpr_test()
{
    constexpr std::string_view cbar{cfoo};
    [[maybe_unused]] constexpr std::string_view cbaz = cfoo;
    return 1;
}

TEST(small_string, standard_construction)
{
    seqan3::small_string foo{"hello"};
    std::string_view bar{foo};
    [[maybe_unused]] std::string_view baz = foo;
    static_cast<std::string_view>(foo);

    [[maybe_unused]] constexpr int i = constexpr_test();
}
```

Reply:
@eseiler do/can we have gcc12 cpp23 mode in our CI now?

Reply:
Not in CI, but in nightlies